src/sentry/auth/helper.py

```diff
@@ -337,16 +320,19 @@ def handle_attach_identity(self, member: OrganizationMember | None = None) -> Au
 
         if member is None:
             member = self._get_organization_member(auth_identity)
-        self._set_linked_flag(member)
+        if member is not None:
```

Comment:
Hmm, I'm curious -- why the new conditional? Was member previous always not None but now is, or did the set_linked_flag just gracefully handle None?

Reply:
I had declared `AuthService.handle_new_membership` to return an optional membership at some point but no longer needs to. That trickled down to here and I added the null check assuming the contract had changed. Fixed in 09b79a9a645b716b6d41a1b06b43b8225b80e777